docs(kic): update rewrite-host guide with URLRewrite filter (#7469)
Kong/docs.konghq.com: app/_src/kubernetes-ingress-controller/guides/requests/rewrite-host.md

Reviewer: We could consider changing the name of this file to reflect these changes but then that also requires changes in nav file.

> There are three options to rewrite the default path handling behavior:

Reviewer: I guess this would also have to be changed given that we can have 4 options in 3.2+.

```suggestion
Users have the following options to rewrite the default path handling behavior:
```

Add the `URLRewrite` filter with `path.replaceFullPath` to your `HTTPRoute` rule to rewrite path entirely.

```yaml
...
filters:
- type: URLRewrite
  urlRewrite:
    path:
      type: ReplaceFullPath
      replaceFullPath: /rewritten-path
```

#### Rewriting path prefix

Add the `URLRewrite` filter with `path.replacePrefixMatch` to your `HTTPRoute` rule to rewrite path prefix.
See the [URLRewrite filter documentation](https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io%2fv1.HTTPPathModifier)
for more information.

```yaml
...
rules:
  - matches:
      - path:
          type: PathPrefix # Only PathPrefix path type is supported with URLRewrite filter using path.type == ReplacePrefixMatch.
          value: /old-prefix
    filters:
     - type: URLRewrite
       urlRewrite:
         path:
           type: ReplacePrefixMatch
           replacePrefixMatch: /new-prefix
```

{% endif_version %}

### Rewrite using regular expressions

{:.note}
> This feature is available from {{ site.kic_product_name }} 2.12 and requires the [`RewriteURIs` feature gate](/kubernetes-ingress-controller/{{ page.release }}/reference/feature-gates/) to be activated.

> > It's only available for Ingress.

Reviewer: This is misleading: line 183 contains the example with HTTPRoute. Do we support `konghq.com/rewrite` for `HTTPRoute`s?

Author: Good catch. `HTTPRoute` example should be removed as we do not support `konghq.com/rewrite` as stated in the note.
